### PR TITLE
Improvements specs have been refactored

### PR DESCRIPTION
**Pull request by TomDudley92**

### CODE REVIEW

File: app/forms/nonresident/ImprovementsForm.scala [hunk 1]
```diff
@@ -27,56 +28,55 @@ import uk.gov.hmrc.play.views.helpers.MoneyPounds
 
 object ImprovementsForm {
 
-  def verifyAmountSupplied(data: ImprovementsModel): Boolean = {
+  private def verifyAmountSupplied(data: ImprovementsModel, showHiddenQuestion: Boolean): Boolean = {
     data.isClaimingImprovements match {
-      case "Yes" => data.improvementsAmt.isDefined || data.improvementsAmtAfter.isDefined
-      case "No" => true
+      case "Yes" if showHiddenQuestion => data.improvementsAmt.isDefined || data.improvementsAmtAfter.isDefined
```
Comment (jameshforster) [inline app/forms/nonresident/ImprovementsForm.scala]: does not need to have the additional if statement

File: test/forms/nonResident/ImprovementsFormSpec.scala [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+      "return a valid form with one error" in {
+        form.errors.size shouldBe 1
+      }
+
+      s"return an error message of ${messages.Improvements.noValueSuppliedError} containing the data" in {
+        form.errors.head.message shouldBe messages.Improvements.noValueSuppliedError
+      }
+    }
+  }
```
Comment (jameshforster) [inline test/forms/nonResident/ImprovementsFormSpec.scala]: Needs a test for an empty value where improvementsForm(false) is used
